src/transformers/models/dpt/modeling_dpt.py

```diff
@@ -700,11 +702,18 @@ def forward(self, hidden_states):
 
         fused_hidden_states = []
         # first layer only uses the last hidden_state
-        fused_hidden_state = self.layers[0](hidden_states[0])
+        size = hidden_states[1].shape[2:] if self.config.use_size else None
+        fused_hidden_state = self.layers[0](hidden_states[0], size=size)
+
         fused_hidden_states.append(fused_hidden_state)
         # looping from the last layer to the second
-        for hidden_state, layer in zip(hidden_states[1:], self.layers[1:]):
-            fused_hidden_state = layer(fused_hidden_state, hidden_state)
+        for idx, (hidden_state, layer) in enumerate(zip(hidden_states[1:], self.layers[1:])):
+            size = (
+                hidden_states[1:][idx + 1].shape[2:]
+                if self.config.use_size and idx != len(hidden_states[1:]) - 1
+                else None
+            )
+            fused_hidden_state = layer(fused_hidden_state, hidden_state, size=size)
```

Comment:
Why is the layer behaviour being controlled in the forward pass here? The layers and modules should be set up with their paramterizations in their inits i.e. the `DPTFusionLayer` should know how it's rescaling, rather than having to recalculate it here on every forward pass

Reply:
This is because the `size` is dynamic, and can only be determined by the size of the feature maps, which is inferred when doing a forward pass (since the model supports varying image resolutions).